user-resource/src/main/java/com/andressag/agenda/user/resource/UserResource.java

```diff
@@ -24,25 +24,74 @@
         this.repository = repository;
     }
 
-    @GetMapping
+    @RequestMapping(path = "/users")
     Iterable<UserView> findAll(HttpServletRequest request) {
         final List<UserView> results = new LinkedList<>();
         try {
             repository.findAll().forEach(entity -> results.add(convert(entity)));
         } catch (Exception error) {
-            final FindUserException exception = new FindUserException(error);
-            exception.setIpAddress(request.getRemoteAddr());
-            exception.setErrorTimestamp(LocalDateTime.now());
-            throw exception;
+            propagateException(request, error);
         }
         return results;
     }
 
+    @GetMapping(path = "/users/{code}")
+    UserView findUserById(@PathVariable("code") Long code, HttpServletRequest request) {
+        UserView result = null;
+        try {
+            result = convert(repository.findOne(code));
+        } catch (Exception error) {
+            propagateException(request, error);
+        }
+        return result;
+    }
+
+    @PostMapping(path = "/users")
+    @ResponseStatus
+    UserView createUser(@RequestBody UserEntity user, HttpServletRequest request) {
+        UserView saved = null;
+        try {
+            saved = convert(repository.save(user));
+        }  catch (Exception error) {
+            propagateException(request, error);
+        }
+        return saved;
+    }
+
+    @PutMapping(path = "/users/{code}")
+    UserView updateUser(@RequestBody UserEntity user, HttpServletRequest request) {
+        UserView updated = null;
+        try {
+            updated = convert(repository.save(user));
+        }  catch (Exception error) {
+            propagateException(request, error);
+        }
+        return updated;
+    }
+
+    @DeleteMapping(path = "/users/{code}")
+    @ResponseStatus
```

Comment:
Same here, but the status should `NO_CONTENT` (or something like that - check the javadoc)